src/zimrecreate.cpp

```diff
@@ -31,6 +31,66 @@
 #include "tools.h"
 #include "version.h"
 
+/**
+ * A PatchItem. This patch html and css content to remove the namespcae from the links.
+ */
+class PatchItem : public zim::writer::Item
+{
+    //article from an existing ZIM file.
+    zim::Item item;
+
+  public:
+    explicit PatchItem(const zim::Item item):
+      item(item)
+    {}
+
+    virtual std::string getPath() const
+    {
+      auto path = item.getPath();
+      if (path.length() > 2 && path[1] == '/') {
+        path = path.substr(2, std::string::npos);
+      }
+      return path;
+    }
+
+    virtual std::string getTitle() const
+    {
+        return item.getTitle();
+    }
+
+    virtual std::string getMimeType() const
+    {
+        return item.getMimetype();
+    }
+
+    std::unique_ptr<zim::writer::ContentProvider> getContentProvider() const
+    {
+        auto mimetype = getMimeType();
+        if ( mimetype.find("text/html") == std::string::npos
+          && mimetype.find("text/css") == std::string::npos) {
+            return std::unique_ptr<zim::writer::ContentProvider>(new ItemProvider(item));
+        }
+
+        std::string content = item.getData();
+        // This is a really poor url rewriting to remove the starting "../<NS>/"
+        // and replace the "../../<NS/" by "../" :
+        // - Performance may be better
+        // - We only fix links in articles in "root" path (`foo.html`) and in one subdirectory (`bar/foo.hmtl`)
+        //   Deeper articles are not fixed (`bar/baz/foo.html`).
+        // - We may change content starting by `'../A/` even if they are not links
+        // - We don't handle links where we go upper in the middle of the link : `../foo/../I/image.png`
+        // - ...
+        // However, this should patch most of the links in our zim files.
+        for (std::string prefix: {"'", "\""}) {
+          for (auto ns : {'A','I','J','-'}) {
+            replaceStringInPlace(content, prefix+"../../"+ns+"/", prefix+"../");
+            replaceStringInPlace(content, prefix+"../"+ns+"/", prefix);
+          }
+        }
```

Comment:
One more case where this may go wrong: `../../<NS>/[AIJ-]/restofurl` will be replaced by `restofurl`.

Reply:
Yes, that's why it is a poor algorithm. And there are probably other issue with this.
If we want to do this well, we have to go as in zimchecks : get all links, resolve them to be absolute links, rewrite them as relative links removing the namespaces. (And to be totally sure, we should parse the html, not just get links by looking for `http='<link>'`)

On the other side, zimrecreate is not wildly use (if not use at all), I've create it because I needed to test zim creation without using mwoffliner. No user asked for it and I don't know anybody using it (even in our team).
And situations where the rewrite fails should be very few.

I think we can go with this algorithm for now.
But @kelson42 may have its word to say here.